File: cs/lc73.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        rows = set()
        cols = set()
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if matrix[i][j] == 0:
                    rows.add(i)
                    cols.add(j)

        for i in rows:
            for j in range(len(matrix[0])):
                matrix[i][j] = 0
        
        for i in cols:
            for j in range(len(matrix)):
                matrix[j][i] = 0

File: cs/lc73.cs
public class Solution {
    public void SetZeroes(int[][] matrix) {
        HashSet<int> rows = new HashSet<int>();
        HashSet<int> cols = new HashSet<int>();
        for(int i=0;i< matrix.Length;i++)
        {
            for(int j=0;j< matrix[0].Length;j++)
            {
                if(matrix[i][j] == 0)
                {
                    rows.Add(i);
                    cols.Add(j);
                }
            }
        }

        foreach(int i in rows)
        {
            for(int j=0;j< matrix[0].Length;j++)
            {
                matrix[i][j] = 0;
            }
        }
        
        foreach(int i in cols)
        {
            for(int j=0;j< matrix.Length;j++)
            {
                matrix[j][i] = 0;
            }
        }
    }
}
